sdk/src/main/scala/com/horizen/SidechainWallet.scala

```diff
@@ -242,6 +237,50 @@ class SidechainWallet private[horizen] (seed: Array[Byte],
   }
 
   override def walletSeed(): Array[Byte] = seed
+
+  override def getWalletBox(boxId: Array[Byte]): JOptional[WalletBox] = {
+    walletBoxStorage.get(boxId).orElse(openedWalletBoxStorage.get(boxId)) match {
+      case Some(wb) => JOptional.of(wb)
+      case None => JOptional.empty()
+    }
+  }
+
+  override def getAllWalletBoxes: JList[WalletBox] = {
```

Comment:
such a method leads to idea, that `WalletBox` should be just an interface with  `closedWalletBox` and `openedWalletBox` implementations. Otherwise it's not clear, that inside the result least it's not only an objects of `WalletBox`